specs/merge/beacon-chain.md

```diff
@@ -209,41 +198,36 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    if is_execution_enabled(state, block.body):
+        process_execution_payload(state, block.body, EXECUTION_ENGINE)  # [New in Merge]
 ```
 
-#### Execution payload processing
+### Execution payload processing
 
-##### `process_execution_payload`
+#### `process_execution_payload`
 
 ```python
-def process_execution_payload(state: BeaconState,
-                              execution_payload: ExecutionPayload,
-                              execution_engine: ExecutionEngine) -> None:
-    """
-    Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
-    """
-    if is_transition_completed(state):
-        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
-        assert execution_payload.number == state.latest_execution_payload_header.number + 1
-
-    assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
-
-    assert execution_engine.new_block(execution_payload)
-
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody, execution_engine: ExecutionEngine) -> None:
+    # Verify consistency of the parent hash and block number
+    if is_merge_complete(state):
+        assert body.execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
+        assert body.execution_payload.block_number == state.latest_execution_payload_header.block_number + uint64(1)
+    # Verify timestamp
+    assert body.execution_payload.timestamp == compute_timestamp_at_slot(state, state.slot)
+    # Verify the execution payload is valid
+    assert execution_engine.on_payload(body.execution_payload)
+    # Cache execution payload
     state.latest_execution_payload_header = ExecutionPayloadHeader(
-        block_hash=execution_payload.block_hash,
-        parent_hash=execution_payload.parent_hash,
-        coinbase=execution_payload.coinbase,
-        state_root=execution_payload.state_root,
-        number=execution_payload.number,
-        gas_limit=execution_payload.gas_limit,
-        gas_used=execution_payload.gas_used,
-        timestamp=execution_payload.timestamp,
-        receipt_root=execution_payload.receipt_root,
-        logs_bloom=execution_payload.logs_bloom,
-        transactions_root=hash_tree_root(execution_payload.transactions),
+        parent_hash=body.execution_payload.parent_hash,
+        coinbase=body.execution_payload.coinbase,
+        state_root=body.execution_payload.state_root,
+        logs_bloom=body.execution_payload.logs_bloom,
+        receipt_root=body.execution_payload.receipt_root,
+        block_number=body.execution_payload.block_number,
+        gas_limit=body.execution_payload.gas_limit,
+        gas_used=body.execution_payload.gas_used,
+        timestamp=body.execution_payload.timestamp,
+        block_hash=body.execution_payload.block_hash,
+        transactions_root=hash_tree_root(body.execution_payload.transactions),
     )
```

Comment:
IMHO it would be less verbose to set `execution_payload = body.execution_payload` first, and then use `execution_payload` instead of `body.execution_payload`.

Reply:
I went with an even cleaner option: `payload = body.execution_payload`. Is that good with you? :)